codice/ddf
# DDF-4368 Updating search and result forms without page refresh

#### What does this PR do?
Allows you to see new search and result forms in the grid without refreshing the page
#### Who is reviewing it? 
@Bdthomson @andrewzimmer 
#### Select relevant component teams: 
#### Ask 2 committers to review/merge the PR and tag them here.
@andrewkfiedler
@bdeining
@djblue
#### How should this be tested?
- Search Forms
  - Navigate to the Search Forms page
  - Create a new form
  - Verify that a success message appears saying the form was successfully saved
  - Ensure that you can see the new form in the grid without refreshing the page
  - Click the interactions drop-down
  - Verify that you see _all_ of the following options:
    - Make Default Form
    - Share Form
    - Edit Form
    - Delete Form
  - Make sure you can successfully complete all these actions
  - When editing, change the title and search term
  - After editing and saving, ensure that you see a success message saying the form was successfully saved
  - Refresh the page and verify that all your edits persisted
- Result Forms
  - Navigate to the Result Forms page
  - Create a new form
  - Verify that success message appears saying the form was successfully saved
  - Ensure that you can see the new form in the grid without refreshing the page
  - Click the interactions drop-down
  - Verify that you see _only_ the following options:
    - Share Form
    - Delete Form
  - Make sure you can successfully complete all these actions
  - When editing, change all three fields
  - After editing and saving, ensure that you see a success message saying the form was successfully saved
  - Refresh the page and verify that all your edits persisted

Additionally, try to edit forms (PUT request) with the target metacard ID in the HTTP JSON body different than the ID in the query param, and ensure the request fails
#### Any background context you want to provide?
#### What are the relevant tickets?
[DDF-4368](https://codice.atlassian.net/browse/DDF-4368)
#### Screenshots
#### Checklist:

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.
